tools/cldr-code/src/main/java/org/unicode/cldr/util/XMLSource.java

```diff
@@ -1601,31 +1601,39 @@ public String getLocaleID() {
         static {
             StandardCodes sc = StandardCodes.make();
             Map<String, Set<String>> countries_zoneSet = sc.getCountryToZoneSet();
-            Map<String, String> zone_countries = sc.getZoneToCounty();
-
-            for (int typeNo = 0; typeNo <= CLDRFile.TZ_START; ++typeNo) {
-                String type = CLDRFile.getNameName(typeNo);
+            Map<String, String> zone_countries = sc.getZoneToCountry();
+            List<NameType> nameTypeList =
+                    List.of(
+                            NameType.LANGUAGE,
+                            NameType.SCRIPT,
+                            NameType.TERRITORY,
+                            NameType.VARIANT,
+                            NameType.CURRENCY,
+                            NameType.CURRENCY_SYMBOL,
+                            NameType.TZ_EXEMPLAR);
+            for (NameType nameType : nameTypeList) {
```

Comment:
Here was a loop depending on the numeric values of CLDRFile.LANGUAGE_NAME, etc. The new code lists the enum values explicitly. This static code block is crucial for code fallback, and while studying it for https://unicode-org.atlassian.net/browse/CLDR-17014 it appeared that refactoring would make that ticket easier to understand and solve